### ISSUE TITLE
Vijay — submission deliverables (story / blog / video / slides) & demo prep

### ISSUE THREAD

**Issue by AnshumanAtrey** [AnshumanAtrey/pharmarl#3]
@vijaykota321 — your tracks while Sahil runs Colab training in #2. Comment back with a commit (and the file paths) once each section is done.

## Context: what we are submitting

The judges grade on this rubric (40 / 30 / 20 / 10):

| Weight | Criterion | What this means for our submission |
|---|---|---|
| 40% | Environment Innovation | Already locked: LLM-as-policy + multi-target + held-out JNK3 transfer test. Don't change. |
| 30% | Storytelling & Presentation | **Your job.** README + blog + video + 3min pitch + 2min Q&A. Must be readable in 3-5 min by a non-specialist. |
| 20% | Showing Improvement in Rewards | Sahil's W&B curve + before/after panel — you embed in README and video. |
| 10% | Reward & Pipeline | Already done: composite oracle, Lipinski gate, parse penalty, zero-atom guard, 9 redteam tests. Mention in pitch defense. |

## Hard submission requirements (non-negotiable per judging-criteria.txt)

- [x] OpenEnv latest release
- [x] Training script using Unsloth + HF TRL in Colab — `colab/train_pharmarl.ipynb`
- [ ] **Real training evidence — loss + reward plots committed to repo** (Sahil produces, you embed)
- [ ] **Mini-blog on HF OR <2 min YouTube OR slide deck** — linked from README
- [x] HF Space deployed and serving
- [ ] **README that motivates problem, explains env, shows results** — all `<TODO>` URLs filled
- [x] Valid `openenv.yaml`, Gym-style API, client/server separation, no reserved names

## What judges find compelling (verbatim from participant help guide)

A strong submission shows:
1. Clear environment design
2. Objective reward functions
3. Evidence that the model improved
4. **Prevention against reward hacking** ← we have a strong story here, lean into it
5. Reproducible deployment story
6. Sharp demo

The recommended demo format (use this exact structure for video + pitch):
1. **Baseline model attempt** (untrained Qwen on JNK3)
2. **Reward/verifier output** (TDC oracle scores)
3. **Trained model attempt** (post-GRPO Qwen on same JNK3)
4. **Measurable improvement** (delta + W&B curve)
5. **Safeguards** (Lipinski gate + zero-atom guard + parse penalty + redteam tests)

---

## Track A — README story rewrite (do FIRST, ~1.5h)

Current README needs a Problem → Environment → Results → Why-it-matters narrative. The reader test: **a non-technical judge reads the README in 3-5 min and wants to try it.**

**Required sections in this order:**

1. **Hero block** (3 lines max): what we built, who it's for, one-line angle (LLM-as-policy molecular RL with held-out generalization test)
2. **Problem** (~150 words): why molecular RL needs a fresh angle; gap = LLM-as-policy is missing from the field
3. **Environment** (~200 words + diagram): what the agent observes, acts, gets rewarded for; reset/step/state mechanics; SELFIES + 4 oracles + Lipinski gate
4. **Results** (placeholder until Sahil ships): W&B curve image + held-out JNK3 bar chart + before/after molecule images. Embed plots, not links.
5. **Reward hacking defenses** (~150 words): parse penalty, Lipinski gate, zero-atom oracle guard, 9 redteam tests, multi-component composite. **Judges specifically look for this.**
6. **Reproduce** (3 commands): pip install, validate_stack.py, demo.py
7. **Submission links table**: HF Space, Colab, video, blog, W&B, GitHub
8. **Why this matters** (~100 words): the LLM-as-policy angle + held-out generalization measurement

**Commit as:** `Vijay: README story rewrite + reproduce instructions`

**Stop condition:** README must be readable in 3-5 min. Have someone outside the team read it and tell you in one sentence what we built. If they can't, rewrite.

---

## Track B — Demo episode video footage (do NOW, ~1h, parallel)

Record this **before** training finishes — it's just env mechanics. Splice in the W&B graph + before/after panel later.

```bash
# Record screen during this:
python -m examples.manual_episode --difficulty easy --target DRD2 \
    --env-url https://anshumanatrey-pharmarl.hf.space
```

What to capture:
- SMILES string updating turn by turn
- 2D molecule rendering (use RDKit `Draw.MolToImage`) — call this in a loop and save PNGs
- Reward number ticking up as Lipinski-passing edits accumulate
- TERMINATE → composite reward score

Save the raw clips. Don't edit yet — wait for training data to splice in.

**Commit as:** `Vijay: demo video footage (raw clips for splicing)`

---

## Track C — HF Blog post draft (~1.5h, parallel)

Mini-blog requirement. ~600-800 words on HuggingFace blog. Use `docs/pitch-script.md` as the skeleton, but expand into prose.

**Structure:**
1. Hook (1 paragraph): what we built, novelty
2. The gap (1 paragraph): molRL has 50+ papers but ~zero LLM-as-policy envs
3. Environment design (~200 words): action space, reward composition, curriculum
4. Reward hacking — what we anticipated and stopped (~200 words): this is the section judges will read closely
5. Results (placeholder for numbers): training curve, before/after, held-out JNK3
6. Honest limitations (~100 words): GSK3B and JNK3 are both Ser/Thr kinases → intra-family transfer, not cross-family. **Don't hide this; flag it.**
7. What's next: foundation-model pretraining as future work

**Save as:** `docs/blog-post.md`. Final post goes to https://huggingface.co/blog/anshumanatrey/pharmarl after training finishes.

**Commit as:** `Vijay: HF blog post draft (numbers TBD)`

---

## Track D — Slide deck (~1h, parallel)

5-7 slide deck as backup to the video (some submission forms accept slides instead of video). Tools: Google Slides or Keynote, export to PDF.

**Slides:**
1. Title + team + one-line pitch
2. The gap (LLM-as-policy is missing from molRL)
3. Environment design diagram
4. Reward composition (table or pie)
5. Training curve (placeholder until Sahil ships)
6. Held-out JNK3 result (placeholder)
7. What we'd do with more compute

**Save as:** `docs/pharmarl-slides.pdf` in the repo.

**Commit as:** `Vijay: 5-slide deck for submission backup`

---

## Track E — Submission form dry-run (~30 min)

Find the actual submission form (Devfolio / Cerebral Valley / wherever the org links it). DO NOT submit yet. Open it, list every field in `docs/submission-form-fields.md`:

- Team name + members
- Problem theme (Theme 3.1 + 4)
- HF Space URL
- GitHub URL
- Colab notebook URL
- W&B run URL (Sahil)
- Video URL (you)
- Blog URL (you)
- Slides URL (you)
- Short description (what's the field length?)
- Long description (what's the field length?)

Pre-fill what you can. Identify which fields are missing data → flag in the doc.

**Commit as:** `Vijay: submission form fields documented`

---

## Track F — Q&A defense practice (~45 min, do near the end)

Read `docs/qa-defense.md` end-to-end. Anticipate these tough questions (likely from researcher judges):

1. *"How is this different from REINVENT or MolDQN?"* → LLM-as-policy with structured JSON actions, drop-in for any chat model
2. *"Why DRD2 specifically, isn't that the most-benchmarked target?"* → That's the point — same canonical benchmark, new policy class
3. *"GSK3B → JNK3 is intra-family; you haven't shown cross-family transfer."* → Conceded honestly; we measured what we measured; future work
4. *"Did you control for QED-correlation in the held-out test?"* → Yes — untrained Qwen baseline on same JNK3 oracle is the control
5. *"What's stopping the agent from gaming the reward?"* → Lipinski gate halves composite, parse penalty, zero-atom guard, multi-component composite, 9 redteam tests
6. *"Why 1.5B model?"* → Colab T4 budget; SFT warmup compensates for shallower priors

Write 1-paragraph answers to each in `docs/qa-defense.md` if not already there.

**Commit as:** `Vijay: Q&A defense rehearsal notes`

---

## Once training finishes (Sahil pings you):

Final assembly track — fast turnaround needed:

1. Pull W&B curve screenshot, save to `docs/plots/training-curve.png` — labeled axes (step, mean_reward), 1-line caption in README
2. Pull held-out JNK3 numbers, fill `[BEFORE]` `[AFTER]` `[DELTA]` blanks in: pitch-script.md, blog-post.md, README.md, slides
3. Generate before/after molecule images via `examples/before_after_demo.py` — save to `docs/plots/before_after.png`
4. Splice video clips + W&B + before/after panel → 90s YouTube upload
5. Publish HF blog post → grab URL → paste into README + submission form
6. Final README pass: every `<TODO>` filled
7. Submit before 5 PM Apr 26

---

## Anti-overclaim discipline

The pitch script's "kill-on-sight" list applies to everything you write. Specifically:

- ❌ "AI for any disease" / "cure all disease"
- ❌ "Saving lives from <disease>"
- ❌ "Replaces medicinal chemists"
- ❌ "Learned binding principles in general"
- ❌ "The LLM brings the chemistry knowledge" (priors are shallow at 1.5B)
- ❌ Specific score predictions ("we'll score 80/100")

If you reach for one of these, swap with: *"trains an LLM to do iterative molecular optimization on a canonical benchmark with verifiable reward."*

---

## How to report progress

Comment on this issue when each Track is done. Format:

```
Track A done — commit abc1234. README readable test: passed (Anshuman read it, said "molRL with LLM-as-policy and held-out generalization", which is correct).
```

If blocked, comment with what's blocking and what you've tried. Don't sit silent.
